src/client/scenes/universe/views/starsystemview.cpp

```diff
@@ -1332,7 +1332,22 @@ void SysStarSystemRenderer::DrawOrbit(const entt::entity& entity) {
     //    glm::quat{{0.f, 0, (float)body.axial}});
     // Draw orbit
     orbit_shader->SetMVP(transform, camera_matrix, m_app.Get3DProj());
-    orbit_shader->Set("color", glm::vec4(1, 1, 1, 1));
+
+    //Set the color of each orbit based on its distance from its center body
+    double dis = m_universe.get<common::components::types::Orbit>(entity).semi_major_axis;
+    double max_dis = m_universe.get<common::components::bodies::Body>(ref).SOI;
+    double inc = m_universe.get<common::components::types::Orbit>(entity).inclination;
```

Comment:
You should replace `m_universe.get<common::components::types::Orbit>(entity)` with setting it to a variable above: `auto& orbit = m_universe.get<common::components::types::Orbit>(entity)` and then call the variable.